### PR TITLE
feat(metric-alerts): Adds subscription consumer for metrics [INGEST-623]

### PR DESCRIPTION
**Pull request by ahmedetefy**
This PR adds subscription consumer for metrics (hidden behind a feature flag) to power crash rate alerts. Also, modifies the kafka stream loader for `metrics_counters` and `metrics_sets` to write to the same commit log topic and the same subscription results topic. 
Upon conversations with @fpacifici , we decided that this is the approach to support consumers for a dataset with multiple entities

### CODE REVIEW

**Comment by lynnagara** [comment]
Not sure I get the motivation for making this change now. Why doing alert/subscriptions part before we even have the metrics data to run queries on and before doing the load testing? Ideally we can avoid ever putting metrics on the old subscriptions infrastructure and go straight to the new one.

File: snuba/datasets/storages/metrics.py
Comment (fpacifici) [inline snuba/datasets/storages/metrics.py:66]: No need to make a difference here. Just change the existing kafka stream loader, metrics are not in prod, no need to maintain two ways

**Review by fpacifici — COMMENTED** [review]
@lynnagara, Likely your change will be ready before this will work in prod anyway. I would not be too concerned of having this in the code base as it won't be in prod any time soon.

@ahmedetefy, please hold off on this PR until you have a the whole pipeline working locally.

**Comment by ahmedetefy** [comment]
@fpacifici I have validated that the entire pipeline works locally with some changes ofcourse

File: snuba/datasets/storages/metrics.py
Comment (jjbayer) [inline snuba/datasets/storages/metrics.py:79]: @ahmedetefy just to double check, you said that making this conditional as with sessions is not necessary, right?

https://github.com/getsentry/snuba/blob/b55fe5027878533f065da8d4b3025cb55c8607eb/snuba/datasets/storages/sessions.py#L120-L132

**Review by jjbayer — APPROVED** [review]
Looks good to me, but I'll leave final approval to the code owners.

File: snuba/datasets/storages/metrics.py
Comment (ahmedetefy) [reply snuba/datasets/storages/metrics.py:79]: Yes since it is not in production yet according to @fpacifici

**Review by untitaker — APPROVED** [review]

**Review by fpacifici — APPROVED** [review]

**Pull request merged**
